### DISCUSSION TITLE
Knowledge bases not working ?

### DISCUSSION THREAD

**Discussion by tiennou44** [open-webui/open-webui#6716] [Issues]
I'm testing Open WebUI (with Ollama) with knowledge bases.
I created two knowledge bases: one called "collection 1" with a pdf file named xxx.pdf and the other called "collection 2" with a pdf file named yyy.pdf.
Each PDF file contains completely different information.
When I start a new conversation, I want to summarize "collection 1". I select "collection 1" with #, then I write my prompt, and I therefore obtain a first response, which summarizes the content of the PDF file of "collection 1".
In the same conversation (this is important), I now select "collection 2", again with #, then write the exact same prompt to summarize the "collection 2".
I then get a new answer, but which summarizes "collection 1" and not "collection 2".
If I now create "collection 3" in the knowledge bases with the files xxx.pdf and yyy.pdf, when I request a summary of the file yyy.pdf after having requested the summary of the file xxx.pdf, it is the same problem: I get the summary from the xxx.pdf file, not from the yyy.pdf file.
And when I request the summary of the entire collection, I get the summary of the xxx.pdf file.

**Comment by Centurion-Rome** [comment]
I seems that # annotation are not working, I have jsut created Knowledge Collection and used #. in prompt:
`provide short summary of content of #mycollection`

frontend showed my collection and added it but AI said: 
"Sure! Could you please provide me with the text or information that you would like to be summarized?"

Maybe I am doing something wrong?

**Reply by KreshLaDoge** [reply]
I think, it doest RAG by default, rather than pasting content of the file into your contnet.

```mermaid
graph LR
    A[Upload Documents] --> B{Extract Text};
    B --> C[Chunk Text];
    C --> D[Vectorize Text];
    E[Prompt LLM with prompt + #mycollection] --> F{Create and Vectorize Query for RAG};
    F --> G[Similarity Search];
    G --> H{Include Top N-Matches in Prompt};
    H --> I[LLM Generates Response];
```

So let's say in your documents there were things about birds, so then the final prompt might look something like this:

```
The bird flew high in the sky.
The bird sang a beautiful song.
The bird ate a worm.

Provide short summary of the content of mycollection
```

**Comment by bodzebod** [comment]
In order to include a knowledge base in a chat, you must first write # , then choose the knowledge base in the dropdown list that appears.

On my side, I have created one with many PDF and docx files. But when I ask a question using o1-mini model, it returns either a reponse in chinese or russian cyrillic ! When I switch to a claude model, I get a response in english.

**Comment by gmaOCR** [comment]
I've made some tests too with normativ documents (quite strict written form so)
Even if I use # to quote the collection, impossible to have the informations from the LLM (i ask a specific paragrafer and it answers paragrafer not avalaible)

I've tried to change the content extract engine with Tika. I've tried also local model with Ollama (deepseek 32 et 14B) or external with openIA key. Same result. 
I thought it could be a bad embedding converter so I tried paraphrase-multilingual, same result.

**Comment by shm0rt** [comment]
Same here, tested with Docling v0.11.0 - same inconsistent results.
Specific issue with word counting: When I ask "How many times does word X appear in the text?", the system returns a different (incorrect) count every single time I ask the same question. The results never correspond to the actual word frequency in the knowledge base documents.
Tested scenarios:

- [X] Document added via hashtag (#) → inconsistent results
- [X] Document specified under knowledge and added to model → same issue
- [X] Same question repeated in same conversation → different answers each time
- [X] New conversations → still inconsistent

Environment:

Open WebUI: v0.6.16-cuda
Ollama: 0.9.6
Docling: v0.11.0
Content extraction engine: docling

This makes the knowledge base feature unreliable for any analytical queries requiring accurate text analysis. The inconsistency suggests either document chunking issues or problems with the embedding/retrieval system when processing word frequency queries.

**Comment by guenhter** [comment]
This is interesting. For us this works actually fine. 

When you ask the second question with KnowlegdeBase2, do you start here a new chat or do you just use the same chat where you have asked the first question with KnowlegdeBase1?
